lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -566,14 +605,18 @@ def generate_clone_spec(src_config)
 
     if config[:customization_ips]
       if get_config(:customization_gw)
-        cust_spec.nicSettingMap = config[:customization_ips].split(',').map { |i| generate_adapter_map(i, get_config(:customization_gw)) }
+        cust_spec.nicSettingMap = config[:customization_ips].split(',').map.with_index { |i, index|
+          generate_adapter_map(i, get_config(:customization_gw), mac_list[index])
+        }
       else
-        cust_spec.nicSettingMap = config[:customization_ips].split(',').map { |i| generate_adapter_map(i) }
+        cust_spec.nicSettingMap = config[:customization_ips].split(',').map.with_index { |i, index|
```

Comment:
I didn't undertake a campaign to modify things other than what I touched, and whether this case collapses doesn't change whether or not my code is added.  I'm entirely willing to undertake a review of the existing code, but hadn't done that yet.

I agree they could collapse.